File: backend/internal/handler/dto/project/phase_permission.go
package project

import (
	"time"

	domainUser "github.com/besart951/go_infra_link/backend/internal/domain/user"
	"github.com/google/uuid"
)

type CreatePhasePermissionRequest struct {
	PhaseID     uuid.UUID       `json:"phase_id" binding:"required"`
	Role        domainUser.Role `json:"role" binding:"required,oneof=admin_fzag fzag admin_planer planer admin_entrepreneur entrepreneur"`
	Permissions []string        `json:"permissions"`
	Permission  *string         `json:"permission"`
}

type UpdatePhasePermissionRequest struct {
	PhaseID     *uuid.UUID       `json:"phase_id"`
	Role        *domainUser.Role `json:"role" binding:"omitempty,oneof=admin_fzag fzag admin_planer planer admin_entrepreneur entrepreneur"`
	Permissions *[]string        `json:"permissions"`
	Permission  *string          `json:"permission"`
}

type PhasePermissionResponse struct {
	ID          uuid.UUID       `json:"id"`
	PhaseID     uuid.UUID       `json:"phase_id"`
	Role        domainUser.Role `json:"role"`
	Permissions []string        `json:"permissions"`
	CreatedAt   time.Time       `json:"created_at"`
	UpdatedAt   time.Time       `json:"updated_at"`
}

type PhasePermissionListResponse struct {
	Items []PhasePermissionResponse `json:"items"`
}

File: frontend/src/lib/domain/phase/phasePermission.ts
import type { UserRole } from '../user/index.js';

export interface PhasePermission {
  id: string;
  phase_id: string;
  role: UserRole;
  permissions: string[];
  created_at: string;
  updated_at: string;
}

export interface PhasePermissionListResponse {
  items: PhasePermission[];
}

export interface CreatePhasePermissionRequest {
  phase_id: string;
  role: UserRole;
  permissions: string[];
}

export interface UpdatePhasePermissionRequest {
  phase_id?: string;
  role?: UserRole;
  permissions?: string[];
}
